### PR TITLE
Improvements to foreground skin

### PR DESCRIPTION
**Pull request by paladox**
This patch includes improvements to the foreground skins.

The improvements are

Remove old foundation files. They were files from foundation 4 and were removed in foundation 5.

Includes css improvements.

Adds styles for preferences so Special:Preferences should now look good.

Fix the hide main page css styles. Change h1 to h2.

Update Foreground.skins.php with some changes.

Fix mw-indicators in foreground so the icons now show on the right instead of left.

Also I have switch foregrounds footer to use icons by default which look much better.

Also fixed action button so instead of it being left its bottom which means it drops down just like it did in foreground 1.2 but it is a little to the right and the indicator above it is on the left instead of right.

Also text that is shown under the titles page is hidden by default since for example it shows like this from Wikipedia but on the footer it already has links for Wikipedia and the wiki name has the name of the wiki.

Re add support for foundation sections which were removed in foundation 5
in favour of tabs.

We carn't use tabs because mediawiki doesn't allow <a> inside wikitext so might as well add support for sections again.

Many more improvements.

### CODE REVIEW

**Comment by paladox** [comment]
@Hutchy68 could you review please.

File: Foreground.skin.php [hunk 1]
```diff
@@ -152,36 +155,42 @@ public function execute() {
...
-		<div id="mw-js-message" style="display:none;"></div>
+			<div id="mw-js-message" style="display:none;"></div>
 
-		<div class="row">
+			<div class="row">
 				<div id="p-cactions" class="large-12 columns">
-					<?php if ($wgUser->isLoggedIn() || $wgForegroundFeatures['showActionsForAnon']): ?>
-						<a id="actions-button" href="#" data-dropdown="actions" data-options="align:left; is_hover: true; hover_timeout:700" class="button small secondary radius"><i class="fa fa-cog"><span class="show-for-medium-up">&nbsp;<?php echo wfMessage( 'actions' )->text() ?></span></i></a>
+					<?php if ($wgUser->isLoggedIn() || $wgForegroundFeatures['showActionsForAnon']) { ?>
+						<a id="actions-button" href="#" data-dropdown="actions" data-options="align:bottom; is_hover: true; hover_timeout:500" class="button dropdown small secondary radius"><i class="fa fa-cog"><span class="show-for-medium-up">&nbsp;<?php echo wfMessage( 'actions' )->text() ?></span></i></a>
```
Comment (Hutchy68) [inline Foreground.skin.php]: still breaking with the dropdown not reorienting to the right.

dropdownalignment

Even if we drop the max-width to 150px it still doesn't look right

narrow150px

I really want it to be a drop under aligned correctly, but seems as if Zurb is on to 6 and 5 is a memory now. I spent many hours trying different things and researching for the fix. The callback in js is just broken detecting the right edge.

File: assets/stylesheets/foreground.css [hunk 2]
```diff
@@ -522,7 +523,7 @@ button, .button {
 }
 
 .top-bar .button.search {
-	top: 0px;
+	top: -1px;
```
Comment (Hutchy68) [inline assets/stylesheets/foreground.css]: Why did you add a -1px, that breaks alignment.

searchbuttonalignmenttop

File: assets/stylesheets/foreground.css [hunk 3]
```diff
@@ -713,13 +714,13 @@ div.mw-htmlform-field-HTMLCheckField div.mw-label { height:0; }
 
 
 /* Jamie's Header Hider */
```
Comment (Hutchy68) [inline assets/stylesheets/foreground.css]: Been kind of wondering if this should be default css or allow the user to make the choice of whether to show the title attributes on the main page or use their MediaWiki:Foreground.css or Common.css to hide it.

File: assets/stylesheets/foreground.css [hunk 4]
```diff
@@ -884,4 +885,129 @@ body.mw-special-Userlogin h2.title {
...
+}
+.htmlform-tip {
+  font-size: x-small;
+  padding: .2em 2em;
+  color: #666;
+}
+fieldset legend {
+  background: none !important;
+}
+
```
Comment (Hutchy68) [inline assets/stylesheets/foreground.css]: preferences better :+1:  I have some code I think we can merge together for preferences improve it more on mobile too.

**Comment by Hutchy68** [comment]
Good PR, just need to have a discussion about the stupid Foundation regression and where to place the dropdown.

**Comment by paladox** [comment]
Ok. Thanks. The reasons I did that to search button is it looked a little of on internet explorer. But I can put it back. The button should also be next to the serve bar not away from it.

**Comment by paladox** [comment]
Did the action button do that on a small screen like and mobile and tablet if so then maybe with css we could move it left a little. But the pic for https://cloud.githubusercontent.com/assets/3527469/12052926/098fb1a6-aee2-11e5-81e3-64f096ecd5f8.PNG looks good we should on set that width in we are on a small screen such as a mobile or small tablet.

File: Foreground.skin.php (hunk 1 above)
Comment (paladox) [inline Foreground.skin.php]: Is there any way on only setting align left for small screens and the rest could be align bottom.

File: assets/stylesheets/foreground.css (hunk 3 above)
Comment (paladox) [inline assets/stylesheets/foreground.css]: Yes. We should probably open an issue to discuss this. If we do decide to let the user decide we should add a notice in read me file on how to hide the main page title. But this cuss only applies to English language wikis since other wikis with different languages won't have this main page title.

File: assets/stylesheets/foreground.css (hunk 4 above)
Comment (paladox) [inline assets/stylesheets/foreground.css]: Ok. Thanks. Would you merge the update to preferences after this patch has been merged.

Comment (Hutchy68) [inline assets/stylesheets/foreground.css]: Yep

**Comment by Hutchy68** [comment]
Actions button does it on medium and up. Small actually works.  We are going to have to face facts it is broken in foundation and it probably won't be fixed. 5.1 - 5.2 not a problem. Appeared in 5.3 claimed fixed in 5.5 but not.  Last comment was in September 2015. 6  is now considered the supported version.

**Comment by Hutchy68** [comment]
Search button was always away even in foreground 1.x.  Because on mobile devices you don't want to fat finger the button and activate the form again.

**Comment by Hutchy68** [comment]
And iOS devices round off the ends of the form for search

**Comment by paladox** [comment]
Maybe we could created a custom drop down js file of foundation with the parts that work and the rest the updated code in 5.5

**Comment by paladox** [comment]
I will upload a patch in a few mins that I have fixed the action button. I made a customisation of the drop down file and css file so upgrading to foundation 6 shoulden affect anything since we presume the problem is fixed in that version. This I fix I have done for it only affects foundation 5.

**Comment by paladox** [comment]
The search button looks out of alignment with the 0px for the button instead of -1px.

And this is what the drop down box looks like for action button.

**Comment by paladox** [comment]
Also it says this part in the changelog Foundation Tabs have a coding change to build tabs on pages but tabs wasn't added in the change that was done in develop branch. So I think that part should be removed in a new pull.

**Comment by paladox** [comment]
@Hutchy68 could you re review this patch since I have re added support for sections. It is not complete but I am not sure what else to do it looks complete too me but it isent't generating the padding-top from js so the content is below the tabs. I have to manualy add padding-top: 55px; to the div so the content is below the tabs. Also left: is not being generated meaning I have to also add that manually.

Please see an example on how the tabs look at

http://simple-random-wikisaur.tk/index.php?title=HHH

I have updated the website template at http://simple-random-wikisaur.tk/index.php?title=Template:Website for the changes so wikiapriary would find it easisar to update.

We would need a follow-up patch that documents the changes.

**Comment by paladox** [comment]
@Hutchy68 I keep getting these js errors in the section file Object doesn't support property or method 'outerWidth' and Object doesn't support property or method 'outerHeight'

**Comment by paladox** [comment]
Could I have help to fix all these issues todo with section please.

**Comment by paladox** [comment]
Ive fixed the padding problem now js will correctly add the padding-top but left is now the only problem too fix.

**Comment by paladox** [comment]
@Hutchy68 I have now fixed both left and padding-top problems this is now ready to be merged.

**Comment by paladox** [comment]
@Hutchy68 please could you review.

**Comment by paladox** [comment]
@Hutchy68 please could you review.

**Comment by paladox** [comment]
@Hutchy68 please could you review and merge.
